### ISSUE TITLE
Add Functionality for non-DM user to direct/control camera [feat] 

### ISSUE THREAD

**Issue by tregolani** [sPOiDar/fvtt-module-stream-view#67]
This is not an issue with the module, as it works wonderfully as intended, but rather a request I would like to make based on the way we use the module. 

I personally record all of our sessions using the "dummy" user method as instructed, but our DM has more than enough on their plate to be worrying about moving and directing the scene. Since the automated camera only follows Friendly/PC actors and tokens, I've often found that the dramatic reveal of an enemy token on another part of the screen ends up being lost in the recording. 

As a player, I would love to be able to direct the streamview client with the DM tools (disable auto, focus on a token, etc) from my PC user account, mainly because moving the camera manually on the streamview user often results in "yoyo-ing" when a PC moves their token and it tries to revert to the automated view. 

I don't know if this is feasible with the current permission system Foundry has for tokens, but I think creating a setting where a specific user account can be designated as the "camera-controller" would be really useful, at least to me!

**Comment by sPOiDar** [comment]
> moving the camera manually on the streamview user often results in "yoyo-ing"

Yeah, StreamView is not intended to be used this way, the stream user is intended to be "view-only".

I'm not saying no to this, but this change would require a pretty significant rearchitecting of the module - currently all code/operations/communication is separated by user class. I think you'd also probably run into issues with token ownership in some scenarios.

**Comment by tregolani** [comment]
I really appreciate your thorough response! It's sounds like the best option for now is just educating my DM on the "focus camera" button going forward. 

That, or running a THIRD window so I can play, record, and direct the camera for the DM 😅

**Comment by TheSabotender** [comment]
you could add a setting so that the camera focuses on Pings for X seconds before returning to its normal position. The setting could also allow this to happen for GM only, or also players. Maybe even toggles for allowing it in and out of combat (only on their turn? or at any time).

While not direct control, it would at least allow players to ping details and cool stuff they want captured on camera :)

**Comment by sPOiDar** [comment]
> you could add a setting so that the camera focuses on Pings for X seconds before returning to its normal position. The setting could also allow this to happen for GM only, or also players

Aren't there core permissions for camera move on ping? Or am I misremembering and that was from a module?
